src/autopr_caller.py

```diff
@@ -0,0 +1,213 @@
+    return message
+
+
+def get_message_body_to_be_sent_to_autopr_queue(
+    branch_name_prefix,
+    edited_files,
+    pr_body,
+    pr_title,
+    repo_owner="dummyUser812",
+    repo_name="deep-learning-containers",
+):
+    message_body = {
+        "repo_owner": repo_owner,
+        "repo_name": repo_name,
+        "branch_name": branch_name_prefix,
+        "edited_files": edited_files,
+        "pr_body": pr_body,
+        "pr_title": pr_title,
+    }
+    return message_body
+
+
+def generate_edited_files_data(image_list, bucket=S3_BUCKET, folder="temp"):
+    edited_files_data = []
+    session = boto3.Session()
+
+    # Then use the session to get the resource
+    s3_resource = session.resource("s3")
+    s3_client = s3_resource.meta.client
+
+    my_bucket = s3_resource.Bucket(S3_BUCKET)
+
+    for s3_object in my_bucket.objects.filter(Prefix=folder):
+        response = s3_client.get_object_tagging(
+            Bucket=bucket,
+            Key=s3_object.key,
+        )
+        tag_set_dict = get_tag_set_dictionary_from_response(response["TagSet"])
+        upload_path = tag_set_dict.get("upload_path", "")
+        truncated_upload_path = upload_path.split("deep-learning-containers/")[-1]
+        image_uri_corresponding_to_the_file = tag_set_dict.get("image_uri", "")
+        if image_uri_corresponding_to_the_file not in image_list:
+            continue
+        if truncated_upload_path:
+            edited_files_data.append(
+                {
+                    "s3_bucket": bucket,
+                    "s3_filename": s3_object.key,
+                    "github_filepath": truncated_upload_path,
+                }
+            )
+
+    return edited_files_data
+
+
+def send_message_to_queue(queue_name, queue_region, message_body_string):
+    sqs = boto3.resource("sqs", region_name=queue_region)
+    queue = sqs.get_queue_by_name(QueueName=queue_name)
+    queue.send_message(MessageBody=message_body_string)
+
+
+def main():
+    from test import test_utils
+
+    dlc_images = test_utils.get_dlc_images()
+    image_list = dlc_images.split(" ")
+    if not image_list:
+        LOGGER.info(f"No image in image_list: {image_list}")
+        return
+    edited_files_data = generate_edited_files_data(
+        image_list=image_list, folder=os.getenv("CODEBUILD_RESOLVED_SOURCE_VERSION", "temp")
+    )
+    ## TODO: Skip in case of no edited files
+    common_image_specs = get_common_image_specs_for_all_images(image_list=image_list)
+    branch_name_prefix = generate_branch_name_prefix(common_image_specs)
+    pr_title = get_pr_title(common_image_specs)
+    pr_body = get_pr_body()
+    ## TODO: Change repo-owner
+    message_body_to_be_sent_to_autopr_queue = get_message_body_to_be_sent_to_autopr_queue(
+        branch_name_prefix=branch_name_prefix,
+        edited_files=edited_files_data,
+        pr_body=pr_body,
+        pr_title=pr_title,
+        repo_owner="dummyUser812",
+        repo_name="deep-learning-containers",
+    )
```

Comment:
Is this something that needs to be changed before the PR is merged?

Reply:
Thanks for pointing this out. This will be changed in a follow up PR once we have set up the bot repo. Till warmup period, we will rely on this content.